sklearn/utils/testing.py

```diff
@@ -398,6 +405,81 @@ def assert_raise_message(exception, message, function, *args, **kwargs):
         assert_in(message, error_message)
 
 
+def assert_not_raises(given_exception, function, *args, **kwargs):
+    """Helper function to make sure that the given error was not raised"""
+
+    try:
+        function(*args, **kwargs)
+    except Exception as e:
+        if isinstance(e, given_exception):
+            raise AssertionError("%r was raised" % e)
+
+
+def _assert_same_model_method(method, X, estimator1, estimator2, msg=None):
+    if hasattr(estimator1, method):
+        m = '%r\n\nhas %s, but\n\n%r\n\ndoes not' % (estimator1,
+                                                     method,
+                                                     estimator2)
+
+        if not hasattr(estimator2, method):
+            raise AttributeError(m)
+
+        # Check if the method(X) returns the same for both models.
+        res1 = getattr(estimator1, method)(X)
+        res2 = getattr(estimator2, method)(X)
+        same_model = (res1.shape == res2.shape) and np.allclose(res1, res2)
+
+        if msg is None:
+            msg = ("Models are not equal. \n\n%s method returned"
+                   " different results:\n\n%r\n\n and\n\n%r\n\n for :\n\n%r"
+                   "\n\nand :\n\n%r" % (method, res1, res2,
+                                        estimator1, estimator2))
+
+        assert same_model, msg
+
+
+def _assert_not_same_model_method(method, X, estimator1, estimator2, msg=None):
+    if hasattr(estimator1, method):
+        m = '%r\n\nhas %s, but\n\n%r\n\ndoes not' % (estimator1,
+                                                     method,
+                                                     estimator2)
+
+        if not hasattr(estimator2, method):
+            raise AttributeError(m)
+
+        # Check if the method(X) returns the same result for both models.
+        res1 = getattr(estimator1, method)(X)
+        res2 = getattr(estimator2, method)(X)
+        not_same_model = (res1.shape != res2.shape) or not np.allclose(res1,
+                                                                       res2)
+
+        if msg is None:
+            msg = ("Models are equal. \n\n%s method returned"
+                   " same results: \n\n%r\n\nfor :\n\n%r \n\nand :\n\n%r" %
+                   (method, res1, estimator1, estimator2))
+
+        assert not_same_model, msg
+
+
+def assert_same_model(X, estimator1, estimator2, msg=None):
```

Comment:
Done.... ( kind of ) Could you please take a look and let me know if those are sufficient?